Framework/Core/src/BoostOptionsRetriever.cxx

```diff
@@ -32,53 +35,62 @@ BoostOptionsRetriever::BoostOptionsRetriever(std::vector<ConfigParamSpec> &specs
     switch(spec.type) {
       case VariantType::Int:
       case VariantType::Int64:
-        options = options(name, bpo::value<int>(), help);
+        options = options(name, bpo::value<int>()->default_value(spec.defaultValue.get<int>()), help);
```

Comment:
here I'm not sure. Default values are optional, so setting the default in the bpo should be done only when we have the default in the option spec.

Reply:
Mmm... True, however if we don't specify one and we dereference it, we will get an exception as we do not return an `std::optional` when we do `get<int>`. Maybe it's time to move to C++17...

Reply:
Actually at the moment there is not way not to specify a default. In case you do not include it, the default constructor is used. We might have to rethink this in the future, but for now this is the correct implementation.